src/Serilog.Sinks.EventLog/Sinks/EventLog/EventLogSink.cs

```diff
@@ -31,7 +31,7 @@ namespace Serilog.Sinks.EventLog
     public class EventLogSink : ILogEventSink
     {
         const string ApplicationLogName = "Application";
-        const int MaximumPayloadLengthChars = 31839;
+        const int MaximumPayloadLengthChars = 31639;
```

Comment:
The [EventLog.WriteEntry](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.eventlog.writeentry) documentation says
> The message string is longer than **31,839** bytes (32,766 bytes on Windows operating systems before Windows Vista).

Is this incorrect?

Reply:
After running the `UsingSuperLongLogMessageWorks` test repeatedly while tweaking the `MaximumPayloadLengthChars` value I figured the actual max length is 31718 on Windows 10. Can you confirm this observation?

Reply:
@0xced This PR would be a very nice improvement so I jumped in and ran the `UsingSuperLongLogMessageWorks` test too, Win 11 22H2. 

The projects dev branch the actual max length is 31701 for me. On mcolleras PR it only worked up to 31698.  
Not sure if this is helpful or just making it worse. :)

Reply:
🤦‍♂️ Nobody seems to know what the actual limit is, see also https://github.com/dotnet/dotnet-api-docs/pull/9999#issuecomment-2316603745

Not sure what's the best course of action.